src/signature.c

```diff
@@ -107,7 +108,7 @@ static ENGINE *get_pkcs11_engine(GError **error)
 	}
 
 	env = g_getenv("RAUC_PKCS11_PIN");
-	if (env != NULL) {
+	if (env != NULL && strlen(env)) {
```

Comment:
This should work as well and avoid the string.h include.

```suggestion
	if (env != NULL && env[0]) {
```

Reply:
maybe more explicit with

```suggestion
	if (env != NULL && env[0] != '\0') {
```

?